text/0000-mixed-shadow-mode.md

```diff
@@ -0,0 +1,196 @@
+evaluates to `false` in mixed mode. This is due to the fact that the synthetic shadow polyfill
+patches and replaces `ShadowRoot`. This will be remediated by customizing the `instanceof` behavior
+using `Symbol.hasInstance`.
+
+### Listening for non-composed events above the root LWC node
+
+There currently exists logic that allows listeners to handle non-composed events outside of the root
+LWC node if the event originates from a non-LWC component in the subtree. This behavior exists for
+legacy reasons and cannot be preserved for components that prefer native Shadow DOM.
+
+### Accessibility
+
+The most common accessibility issue in LWC is related to id-referencing across shadow boundaries.
+The current workaround for this is to bypass the LWC engine's id-mangling by setting attributes
+dynamically, but such a workaround would not work in native Shadow DOM.
+
+### Instrumentation
+
+Applications that rely on instrumentation libraries that don't yet support Shadow DOM are currently
+able to obtain references to descriptors like `addEventListener()` before they are patched and use
+them to override `@lwc/synthetic-shadow` polyfills where needed. Such workarounds would not work for
+events originating from components that prefer native Shadow DOM.
+
+### Styling
+
+Applications currently rely on global stylesheets to apply deeply throughout the DOM. With native
+Shadow DOM, this would not be possible. All of a component's styles would have to be generated and
+added to the component bundle.
+
+## Drawbacks
+
+A potential (non-verified) drawback is that existing components may experience a slight performance
+degradation due to the fact that mixed mode requires logic-forking whenever a Shadow DOM API is
+invoked.
+
+## Alternatives
+
+Other alternatives have not been considered but contributions to this proposal are welcome.
+
+## Adoption strategy
+
+Component authors must opt-in when implementing components that prefer native shadow in a mixed
+shadow DOM environment due to observable differences between synthetic mode and native mode that are
+not statically verifiable. As such, identifying and resolving for any breaking changes when an
+existing component signals that it prefers native Shadow DOM would be the responsibility of the
+component.
+
+Note that the current proposal for mixed mode considers polyfills as all-or-nothing.  LWC is not in
+the business of providing workarounds for discrepancies in browser implementations, unless it
+affects the functionality of the framework itself.
+
+# How we teach this
+
+Developers should know about mixed mode and the ability for a component to favor native Shadow DOM
+before they start implementing new components. This would eliminate any potential tech debt before
+they get started. Any unresolved observable differences should be documented to assist in deciding
+whether a component is a candidate for native Shadow DOM.

```

Comment:
Thinking about the developer experience, I wonder if it would be useful to offer customers a way to enable native shadow DOM mode globally for all their components. You can imagine that some shop may eagerly migrate all their components to use `preferNativeShadow = true`, but now they can't ensure that any _new_ components will be written in native shadow. And for developers, it's easy to forget to add the static property. Whereas if they could control the default value, then they could ensure that they're futureproof before we switch the framework default from `false` to `true`.

That said, I'm not sure technically how to achieve this, or how to scope the flag properly.

Reply:
I think that in the case you described, they could just remove the synthetic shadow polyfill.

Reply:
I'm thinking of this from the perspective of, say, a managed package author. If someone is off-core and owns the whole stack, then yeah, they can just remove the polyfill.

Reply:
An alternative solution would be to introduce a new `prefer-native-shadow` linting rule. This linting rule could check if all the classes extending from `LightningElement` have a `static preferNativeShadow = true`.